doc/elements-tx-format.md

```diff
@@ -0,0 +1,562 @@
+|-----|--------|-----|---------|--------|-----|
+| Asset | Yes | 33 bytes | `ConfidentialAsset` | | Cannot be null. |
+| Amount | Yes | 9 or 33 bytes | `ConfidentialAmount` | | Cannot be null. |
+| Nonce | Yes | 1 or 33 bytes | `ConfidentialNonce` | | |
+| ScriptPubKey Length | Yes | Varies | `VarInt` | | | |
+| ScriptPubKey | If ScriptPubKey Length is non-zero | Varies | `hex` | Big-endian | |
+
+#### Witness
+
+As noted above, the witness is only present for SegWit transactions.
+
+|Field|Required|Size|Data Type|Encoding|Notes|
+|-----|--------|-----|---------|--------|-----|
+| Input Witnesses| Yes | Varies | `Vector<InputWitness>` | | There is exactly one input witness for each input in the transaction.<br><br>This number is not explicitly included in the witness — it is implied by the number of inputs. |
+| Output Witnesses| Yes | Varies | `Vector<OutputWitness>` | | There is exactly one output witness for each output in the transaction.<br><br>This number is not explicitly included in the witness — it is implied by the number of outputs. |
+
+#### InputWitness
+
+SegWit transactions have one such witness for each input.
+
+|Field|Required|Size|Data Type|Encoding|Notes|
+|-----|--------|-----|---------|--------|-----|
+| Issuance Amount Range Proof | Yes | Varies | `Proof` | | `0x00` → null. |
+| Inflation Keys Range Proof | Yes | Varies | `Proof` | | `0x00` → null. |
+| Script Witness | Yes | Varies | `Vec<hex>` | | The vector represents the witness stack.<br>Can be empty (length of 0). |
+| Pegin Witness | Yes | Varies | `Vec<hex>` | | The vector represents the witness stack.<br>Can be empty (length of 0). |
+
+The range proofs must be empty if their asociated amounts (issuance / inflation keys) are explicit.
+
+#### OutputWitness
+
+SegWit transactions have one such witness for each output.
+
+|Field|Required|Size|Data Type|Encoding|Notes|
+|-----|--------|-----|---------|--------|-----|
+| Surjection Proof | Yes | Varies | `Proof` | | Will be null (`0x00`) if corresponding output is not confidential. |
+| Range Proof | Yes | Varies | `Proof` | | Will be null (`0x00`) if corresponding output is not confidential. |
+
+The Range Proof must be empty if the output’s amount is explicit. 
+
+The Surjection Proof must be empty if the output’s asset ID is explicit.
+
+#### AssetIssuance
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Asset Blinding Nonce | Yes | 32 bytes | `hex` | | Zero for a new asset issuance; otherwise a blinding factor for the input. |
+| Asset Entropy | Yes | 32 bytes | `hex` | | **New issuances:** Freeform entropy field, no consensus-defined meaning, but is used as additional entropy to the asset tag calculation.<br><br>**Reissuances:** Required to be the asset's entropy value (from its initial issuance). |
+| Amount | Yes | 1 or 9 or 33 bytes | `ConfidentialAmount` | | Amount of the asset to issue. Both explicit and blinded amounts are supported. |
```

Comment:
Note: Amount and Num Inflation Keys cannot be explicitly 0.  (Instead they should be NULL values).